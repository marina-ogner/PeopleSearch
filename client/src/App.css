.center{
    margin: 2% auto;
    text-align: center;
}
label{
    margin-left:15px;
    margin-right:5px;
}
#button{
    margin-left:15px;
    margin-bottom:15px;
}
@media only screen and (max-width: 1000px){
    .sm-0{
        display: block;
    }
}
.grey{
    padding:10px;
    border-bottom: 1px solid rgb(201, 199, 199);
}
.containerMy{
    margin:10px;
    padding:0px;
    border: 1px solid rgb(201, 199, 199);
}
.alertMy{
    width:50%;
    margin: 10px auto;
    text-align: center;
}
.inputError{
color: red;
text-align: right;
height: 15px;
}