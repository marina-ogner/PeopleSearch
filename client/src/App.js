import React from 'react';
import './App.css';
import Titles from "./components/Titles";
import Form from "./components/Form";
import People from "./components/People";
import fetch from 'isomorphic-fetch';

class App extends React.Component {

  state = {
    people: [],
    error: undefined,
    loading: false
  }

  getPeople = async (e) => {
    e.preventDefault();
    const firstName = e.target.elements.firstName.value;
    const lastName = e.target.elements.lastName.value;
    const peopleState = e.target.elements.state.value;
    if (firstName.trim().length > 0 && lastName.trim().length > 0 && peopleState.trim().length > 0) {
      this.setState({
        people: [],
        error: undefined,
        loading: true
      })
      const api_call = await fetch(`api/${firstName}/${lastName}/${peopleState}`);
      const data = await api_call.json();
      if (data.datafinder.results) {
        this.setState({
          people: data.datafinder.results,
          error: "",
          loading: false
        })
      }
      else {
        this.setState({
          people: [],
          error: "No result Found",
          loading: false
        });
      }
    }
    else {
      this.setState({
        people: [],
        error: undefined,
        loading: false
      })
    }
  }
  render() {
    return (
      <div className="container">
        <div className="center">
          <Titles />
        </div>
        <Form load={this.state.loading} getPeople={this.getPeople} />
        <People
          load={this.state.loading}
          error={this.state.error}
          people={this.state.people.map(person => ([person.FirstName, person.LastName, person.Address, person.City, person.State]))}
        />
      </div>
    );
  }
};
export default App;



