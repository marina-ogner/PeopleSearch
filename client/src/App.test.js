import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('App component should render as expected', () => {
  const component = shallow(<App />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
})

// only can test if the line in the App.js changed to const api_call = await fetch(`http://localhost:5000/api/${firstName}/${lastName}/${peopleState}`);
// but it creates CORS problem:
// test('App component should clear the result section and start loading process with all valid inputs', () => {
//   const component = shallow(<App />);
//   const e = { target: { elements: { firstName: { value: "MARK" }, lastName: { value: "PRENTISE" }, state: { value: "WA" } } }, preventDefault: () => { } };
//   component.instance().getPeople(e);
//   expect(component.state().people.length).toBe(0);
//   expect(component.state().error).toBe(undefined);
//   expect(component.state().loading).toBe(true);
//   const tree = toJson(component);
//   expect(tree).toMatchSnapshot();
// });
// test('App component should clear the result section with even one not valid input', () => {
//   const component = shallow(<App />);
//   const e = { target: { elements: { firstName: { value: "" }, lastName: { value: "PRENTISE" }, state: { value: "WA" } } }, preventDefault: () => { } };
//   component.instance().getPeople(e);
//   expect(component.state().people.length).toBe(0);
//   expect(component.state().error).toBe(undefined);
//   expect(component.state().loading).toBe(false);
//   const tree = toJson(component);
//   expect(tree).toMatchSnapshot();
// });
// test('App component should clear the result section with even one not valid input', () => {
//   const component = shallow(<App />);
//   const e = { target: { elements: { firstName: { value: "MARK" }, lastName: { value: "" }, state: { value: "WA" } } }, preventDefault: () => { } };
//   component.instance().getPeople(e);
//   expect(component.state().people.length).toBe(0);
//   expect(component.state().error).toBe(undefined);
//   expect(component.state().loading).toBe(false);
//   const tree = toJson(component);
//   expect(tree).toMatchSnapshot();
// });
// test('App component should clear the result section with even one not valid input', () => {
//   const component = shallow(<App />);
//   const e = { target: { elements: { firstName: { value: "MARK" }, lastName: { value: "PRENTISE" }, state: { value: "" } } }, preventDefault: () => { } };
//   component.instance().getPeople(e);
//   expect(component.state().people.length).toBe(0);
//   expect(component.state().error).toBe(undefined);
//   expect(component.state().loading).toBe(false);
//   const tree = toJson(component);
//   expect(tree).toMatchSnapshot();
// });
// test('App component should clear the result section with even one not valid input like "  "', () => {
//   const component = shallow(<App />);
//   const e = { target: { elements: { firstName: { value: "  " }, lastName: { value: "PRENTISE" }, state: { value: "WA" } } }, preventDefault: () => { } };
//   component.instance().getPeople(e);
//   expect(component.state().people.length).toBe(0);
//   expect(component.state().error).toBe(undefined);
//   expect(component.state().loading).toBe(false);
//   const tree = toJson(component);
//   expect(tree).toMatchSnapshot();
// });
// test('App component should clear the result section and start loading process even with input like "  MARK "', () => {
//   const component = shallow(<App />);
//   const e = { target: { elements: { firstName: { value: "  MARK " }, lastName: { value: "PRENTISE" }, state: { value: "WA" } } }, preventDefault: () => { } };
//   component.instance().getPeople(e);
//   expect(component.state().people.length).toBe(0);
//   expect(component.state().error).toBe(undefined);
//   expect(component.state().loading).toBe(true);
//   const tree = toJson(component);
//   expect(tree).toMatchSnapshot();
// });
