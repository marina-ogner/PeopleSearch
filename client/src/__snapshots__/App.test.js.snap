// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render as expected 1`] = `
<div
  className="container"
>
  <div
    className="center"
  >
    <Titles />
  </div>
  <Form
    getPeople={[Function]}
    load={false}
  />
  <People
    load={false}
    people={Array []}
  />
</div>
`;
